packages/muelu/src/Graph/MatrixTransformation/MueLu_UnsmooshFactory_def.hpp

```diff
@@ -222,18 +222,21 @@ namespace MueLu {
       }
     }
     Teuchos::RCP<Map> coarseColMap = MapFactory::Build(amalgP->getDomainMap()->lib(),
-        Teuchos::OrdinalTraits<Xpetra::global_size_t>::invalid(),
-        unsmooshColMapGIDs(), //View,
-        indexBase,
-        amalgP->getDomainMap()->getComm());
+                                                       Teuchos::OrdinalTraits<Xpetra::global_size_t>::invalid(),
+                                                       unsmooshColMapGIDs(), //View,
+                                                       indexBase,
+                                                       amalgP->getDomainMap()->getComm());
 
     // Assemble unamalgamated P
-    Teuchos::RCP<CrsMatrix> unamalgPCrs = CrsMatrixFactory::Build(unamalgA->getRowMap(),coarseColMap, 3);
-    for (decltype(rowCount) i = 0; i < rowCount; i++) {
-      unamalgPCrs->insertLocalValues(i, newPCols.view(newPRowPtr[i],newPRowPtr[i+1]-newPRowPtr[i]),
-          newPVals.view(newPRowPtr[i],newPRowPtr[i+1]-newPRowPtr[i]));
+    Teuchos::RCP<CrsMatrix> unamalgPCrs = CrsMatrixFactory::Build(unamalgA->getRowMap(),
+                                                                  coarseColMap,
+                                                                  maxDofPerNode*amalgP->getNodeMaxNumRowEntries());
```

Comment:
If you already have row pointers, why are you using `insertLocalValues` instead of just building the matrix directly?

Reply:
Truth be told I am only attempting to make the code work with Tpetra deprecated code, I did not read the whole class and my guess is someone could optimize it if we have any important use case for it.